src/server/lib/middleware/errorHandler.ts

```diff
@@ -33,6 +33,21 @@ export const routeErrorHandler = (
       ),
     );
     return next(err);
+  } else if (err.code === 'EBADRECAPTCHA') {
+    res.redirect(
+      303,
+      addQueryParamsToPath(
+        req.url,
+        {
+          ...res.locals.queryParams,
+          returnUrl: defaultReturnUri,
```

Comment:
we does returnUrl get overridden?
I don't know why it does this on EBADCSRFTOKEN

Reply:
good question! not sure why. on the face of it I can't see why we'd want to do that in the case of handling the recaptcha error.. although it'd be good to know why it was there for the csrf check

Reply:
@coldlink 36d5feb69c322f58e2792f7218a1f9ccce31ae2d

Reply:
thanks for explaining how this bit is working! Sounds like it's a tried and tested solution, I can make us a ticket to look into this if it doesn't need changing immediately here

Reply:
yeah it will only affect a small number of users who encounter CSRF errors, so we can add something to the backlog to look at it, it's not urgent :)